ValheimPlus/RPC/VPlusConfigSync.cs

```
@@ -61,41 +61,41 @@ public static void RPC_VPlusConfigSync(long sender, ZPackage configPkg)
+                                    var line = configPkg.ReadString();
+                                    tmpWriter.WriteLine(line);
+                                }
 
-                                tmpWriter.WriteLine(line);
-                            }
+                                tmpWriter.Flush(); //Flush to memStream
+                                memStream.Position = 0; //Rewind stream
 
-                            tmpWriter.Flush(); //Flush to memStream
-                            memStream.Position = 0; //Rewind stream
+                                ValheimPlusPlugin.harmony.UnpatchSelf();
+                                Configuration.Current = ConfigurationExtra.LoadFromIni(memStream);
 
-                            ValheimPlusPlugin.harmony.UnpatchSelf();
+                                ValheimPlusPlugin.harmony.PatchAll();
 
-                            // Sync HotKeys when connecting ?
-                            if(Configuration.Current.Server.IsEnabled && !Configuration.Current.Server.serverSyncHotkeys)
-                            {
-                                isConnecting = true;
-                                Configuration.Current = ConfigurationExtra.LoadFromIni(memStream);
-                                isConnecting = false;
+                                ZLog.Log("Successfully synced VPlus configuration from server.");
                             }
-                            else
-                            {
-                                Configuration.Current = ConfigurationExtra.LoadFromIni(memStream);
-                            }
-                                
-                            ValheimPlusPlugin.harmony.PatchAll();
-
-                            ZLog.Log("Successfully synced VPlus configuration from server.");
                         }
                     }
+                    catch (Exception)
+                    {
+                        //ignore
+                    }
```

Are you expecting any exceptions? If not, you probably should omit the catch. Ignoring an error here would lead to only local settings being loaded which could cause surprises when buildings collapse because `StructuralIntegrity` is disabled

---

The code is too dirty to not expect exceptions. The only thing I could do is logging an error. But my intention was to reset always the property SyncRemote, which tells, that the mod is currently syncing. This is important for HotkeySync to load local valies if not synced or local config is loading.

---

I have a full team of developers. Maybe they'll accept my suggestion to revise the code... ;-)

I don't expect the original developers coming back. Your PR is over 2 weeks old. Others are older. I've found comments saying it's discontinued. https://www.reddit.com/r/ModdedValheim/comments/11qh6o3/comment/jcb4qqf/

---

New commit with rethrow of the error.